util/src/main/java/net/automatalib/util/automata/fsa/DFAs.java

```diff
@@ -369,4 +371,30 @@ private DFAs() {
         return HopcroftMinimization.minimizeDFA(dfa);
     }
 
+    /**
+     * Computes whether the given DFA is prefix-closed.
+     *
+     * @param dfa the DFA to check
+     * @param alphabet the Alphabet
+     * @param <S> the type of state
+     * @param <I> the type of input
+     *
+     * @return whether the DFA is prefix-closed.
+     */
+    public static <S, I> boolean isPrefixClosed(DFA<S, I> dfa, Alphabet<I> alphabet) {
+        boolean prefixClosed = true;
+        final List<S> states = new ArrayList<>(dfa.getStates());
+        for (int s = 0; s < states.size() && prefixClosed; s++) {
+            final S state = states.get(s);
+            if (!dfa.isAccepting(state)) {
+                for (int i = 0; i < alphabet.size() && prefixClosed; i++) {
+                    final I input = alphabet.getSymbol(i);
+                    final S next = dfa.getSuccessor(state, input);
+                    prefixClosed = next == null || !dfa.isAccepting(next);
+                }
+            }
```

Comment:
I think both loops can be formulized as for-each loops (by directly returning `false` if you encounter a violating constellation). This way you could omit copying the states and performing the `alphabet.getSymbol(i)`lookup, which may be expensive depending on the alphabet implementation.